libtenzir/src/package.cpp

```diff
@@ -354,6 +377,80 @@ auto package_config::parse(const view<record>& data)
 
 namespace {
 
+template <class Value>
+auto parse_operator_parameter_list(std::string_view field_name,
+                                   const Value& value)
+  -> caf::expected<std::vector<package_operator_parameter>> {
+  auto result = std::vector<package_operator_parameter>{};
+  if (is<caf::none_t>(value)) {
+    return result;
+  }
+  const auto* lst = try_as<view<list>>(&value);
+  if (not lst) {
+    return diagnostic::error("{} must be a list", field_name)
+      .note("invalid package definition")
+      .to_error();
+  }
+  result.reserve(lst->size());
+  for (const auto& elem : *lst) {
+    const auto* rec = try_as<view<record>>(elem);
+    if (not rec) {
+      return diagnostic::error("{} entries must be records", field_name)
+        .note("invalid package definition")
+        .to_error();
+    }
+    TRY(auto param, package_operator_parameter::parse(*rec));
+    result.push_back(std::move(param));
+  }
+  return result;
+}
+
+auto ascii_iequals(std::string_view lhs, std::string_view rhs) -> bool {
+  return lhs.size() == rhs.size()
+         && std::equal(lhs.begin(), lhs.end(), rhs.begin(), rhs.end(),
+                       [](char l, char r) {
+                         return std::tolower(l) == std::tolower(r);
+                       });
+}
+
+auto is_field_path_type(const package_operator_parameter& param) -> bool {
+  return param.type && ascii_iequals(*param.type, "field");
+}
+
+auto normalize_basic_type_name(std::string_view name) -> std::string {
+  if (ascii_iequals(name, "int") || ascii_iequals(name, "integer")) {
+    return "int64";
+  }
+  if (ascii_iequals(name, "count")) {
+    return "uint64";
+  }
+  if (ascii_iequals(name, "float") || ascii_iequals(name, "real")) {
+    return "double";
+  }
+  return std::string{name};
+}
+
+auto parse_parameter_value_type(const package_operator_parameter& param,
+                                std::string_view op_id, diagnostic_handler& dh)
+  -> failure_or<std::optional<type>> {
+  if (! param.type || is_field_path_type(param)) {
+    return std::optional<type>{};
+  }
+  auto normalized = normalize_basic_type_name(*param.type);
+  auto legacy = legacy_type{};
+  auto f = normalized.begin();
+  auto l = normalized.end();
+  if (! parsers::legacy_type(f, l, legacy) || f != l) {
+    diagnostic::error("invalid type '{}' for parameter '{}' in operator '{}'",
```

Comment:
We always use backticks for things related to code. Same goes for all other diagnostics in this diff.